controllers/datadogagent/feature/sbom/feature.go

```diff
@@ -161,6 +165,33 @@ func (f *sbomFeature) ManageNodeAgent(managers feature.PodTemplateManagers, prov
 			Value: strings.Join(f.containerImageAnalyzers, " "),
 		})
 	}
+	if f.containerImageUncompressedLayersSupport {
+		if f.containerImageOverlayFSDirectScan {
+			managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
```

Comment:
Is it intentional that the new env vars are added to only node agent container, while existing ones (`DDSBOMEnabled`, `DDSBOMContainerImageAnalyzers`, etc.) are added to all containers?

Reply:
Test doesn't make the difference clear either. This verifies that previous env vars are added to all containers, there is no test for the new env vars. Could you please add one so the intention is clear?
https://github.com/DataDog/datadog-operator/pull/1201/files#diff-62de4539307e0ac7e109b94bb59b9ad8faff84eae1258be8b76a8c7d7212b523R89-R90

Reply:
Yes, it is intentional. This option currently only makes sense for the core agent.

Reply:
Added a test for it

Reply:
Thanks!